go/mcap/cmd/doctor.go

```diff
@@ -0,0 +1,178 @@
+package cmd
+
+import (
+	"os"
+	"io"
+	"bytes"
+	"fmt"
+	"encoding/binary"
+
+	"github.com/fatih/color"
+	"github.com/foxglove/mcap/go/libmcap"
+	"github.com/spf13/cobra"
+)
+
+type mcapRecordReader struct {
+	reader       io.Reader
+}
+
+type RawRecord struct {
+	OpCode libmcap.OpCode
+	Bytes []byte
+}
+
+func (this *mcapRecordReader) Warn(format string, v ...interface{}) {
```

Comment:
"this" isn't typically used in the same way in go as in js. It would be more idiomatic to call this by a short initial, like `r`, or `rr`.

Also, in terms of structuring the code, I think we will probably want to pull the logging functionality into a separate logging module (next to cmd/) so that all the other pieces can make use of the colors and such - so probably wouldn't attach these methods to mcapRecordReader, but write some `func Errorf(string, ...interface{})`, `func Warnf(string, ...interface{})`, etc with whatever colors we want to attach.


Code would then use those like this:

`import (
log "github.com/foxglove/mcap/go/mcap/logging"
)

log.Warnf("Hello this is red")
```